### PR TITLE
Make CI consider L/ASAN options

### PR DESCRIPTION
**Pull request by johnfxgalea-e**
Adds a suppression list to avoid LSAN from complaining about known false positives. To achieve this, we rely on the settings of LSAN and ASAN options.

### CODE REVIEW

File: lsan_suppression.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Leak arises from OpenSSL usage of MoCOCrW and SoftHSM2.
```
Comment (wusto) [inline lsan_suppression.txt]: This is too brief to judge.
Please add a technical explanation why this leak cannot be solved and is accepted.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -84,7 +89,8 @@ jobs:
 
       - name: "Run tests"
         run: |
-          docker run --rm -e SOFTHSM2_CONF="${{ matrix.softhsm2_conf }}" \
+          docker run --rm -e ASAN_OPTIONS="${{ env.ASAN_OPTIONS }}" -e LSAN_OPTIONS="${{ env.LSAN_OPTIONS }}" \
```
Comment (wusto) [inline .github/workflows/ci.yml]: What happens if ASAN options and LSAN options are provided but the build-type is a no-asan one?

I am not sure, if I like the overloading and somehow tend to create an extra step which is dependent on the build-type.

File: docker-build-env/Dockerfile [hunk 3]
```diff
@@ -11,6 +11,8 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -y --no-install-rec
         googletest \
         libboost-all-dev \
         libssl-dev \
+        # For llvm-symbolizer (used by ASAN).
+        llvm \
```
Comment (wusto) [inline docker-build-env/Dockerfile:17]: Wasn't ASAN already enabled in previous builds?
Is there a reason why it is needed now?

**Review by wusto — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (johnfxgalea-e) [reply .github/workflows/ci.yml]: They are ignored. Do you prefer if then else statements, with and without OPTION specification instead?

File: docker-build-env/Dockerfile (hunk 3 above)
Comment (johnfxgalea-e) [reply docker-build-env/Dockerfile:17]: This is needed for LLVM's symboliser to print clearer stack-traces required for more accurate suppression.

Comment (wusto) [reply docker-build-env/Dockerfile:17]: Do you want to add this to your comment?
Iirc Ivan already asked the same. So it is not obvious.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (wusto) [reply .github/workflows/ci.yml]: yes. That would be great.
Then it is clear what is ran.

**Review by wusto — CHANGES_REQUESTED** [review]

File: docker-build-env/Dockerfile (hunk 3 above)
Comment (johnfxgalea-e) [reply docker-build-env/Dockerfile:17]: Done.

File: lsan_suppression.txt (hunk 1 above)
Comment (johnfxgalea-e) [reply lsan_suppression.txt]: It is not MoCOCwR code, as far as I am aware. Added more clarification to the comment.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (johnfxgalea-e) [reply .github/workflows/ci.yml]: I fixed this with matrix CI code instead, in order to avoid clunky if statements.

File: lsan_suppression.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# Leak arises from OpenSSL p11 and SoftHSM2, where an engine is allocated during initilisation.
```
Comment (istepic) [inline lsan_suppression.txt]: It's not clear where and in what scenario it leaks. A trace would help.

**Review by istepic — APPROVED** [review]
This leak doesn't look problematic to me. Consider fixing/raising issue upstream since MoCOCrW is not using OPENSSL_init_crypto.

File: lsan_suppression.txt (hunk 4 above)
Comment (johnfxgalea-e) [reply lsan_suppression.txt]: Makes sense. Will do!

Comment (johnfxgalea-e) [reply lsan_suppression.txt]: Done.

**Review by wusto — APPROVED** [review]

**Pull request merged**